nautobot/docs/development/react-ui.md

```diff
@@ -0,0 +1,105 @@
+# Building the React UI
+
+## Introduction
+
+Nautobot 2.0 introduces a new user interface built on React. This new UI is a complete replacement for the previous Django-based UI, and is the recommended interface for all users. The Django UI will be removed in a future release.
+
+## Design Patterns
+
+A key change with moving to a client-side application for the user interface is that the application is now stateful. This means that the application is aware of the current state of the user interface, and can update the UI without requiring a page refresh. This is a significant change from the previous Django-based UI, which was stateless. In order to achieve this, a perspective on how the UI is rendered is required. Since much of the data is retrieved from the API, the UI is rendered in two phases:
+
+- **Phase 1**: The UI is rendered with placeholder data, and the API is queried for the actual data
+- **Phase 2**: The UI is re-rendered with the actual data
+
+This encourages the UI to be built in a way that is resilient to changes in the data, and allows the UI to be updated without requiring a page refresh. Placeholder data can either be empty, for example on a list view this is easy: an empty list `[]`. But in more complex cases, a view may choose to show a loading indicator until the data is available.
+
+In most cases, we build the UI with the preference we only display the final component once the data is available and display a loading indicator until then.
+
+### State Management
```

Comment:
Future enhancement idea for some more sections under `State Management`: 

- Reducers